sdk/src/main/java/io/radar/sdk/RadarVerificationManager.kt

```diff
@@ -0,0 +1,60 @@
+package io.radar.sdk
+
+import android.content.Context
+import com.google.android.play.core.integrity.IntegrityManagerFactory
+import com.google.android.play.core.integrity.IntegrityTokenRequest
+
+internal class RadarVerificationManager(
+    private val context: Context,
+    private val logger: RadarLogger,
+) {
+
+    fun getIntegrityToken(googleCloudProjectNumber: Long?, nonce: String?, block: (integrityToken: String?, integrityException: String?) -> Unit) {
+        logger.d("Requesting integrity token")
+
+        if (googleCloudProjectNumber == null) {
+            val integrityException = "Missing Google Cloud project number"
+
+            logger.d(integrityException)
+
+            block(null, integrityException)
+
+            return
+        }
+
+        if (nonce == null) {
```

Comment:
In testing, it looks like this can be an empty string if the server does not return it where expected.